internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -254,6 +353,293 @@ var _ = Describe("cluster reconcile", func() {
 		mockCtrl.Finish()
 	})
 
+	Context("Mirror Registry", func() {
```

Comment:
I'd prefer this be something more like `Describe("processMirrorRegistryConfig", func() {`

That way we can separate the tests that are specifically for this function from the ones that are testing `Reconcile` (which I think should probably be with the other tests).

Generally I don't feel like there's more value in grouping tests by feature rather than the function they're testing.